lib/core.rb

```diff
@@ -11,11 +11,7 @@ def initialize
 
   def guess(number)
     @statistics[:tries] += 1
-    case number.to_i <=> @secret
-    when -1 then :higher
-    when 0 then :win # end of game flag
-    when 1 then :lower
-    end
+    { -1 => :higher, 0 => :win, 1 => :lower }[number.to_i <=> @secret]
```

Comment:
This looks pretty good.  What could the name of the hash be?  Would it be good if it where a constant on the class?

`POTENTIAL_RESULTS[number.to_i <=> @secret]` would be potentially how it could be called.  If the language were changed for the values, then you would not need to dig into the code.